python/ray/data/iterator.py

```diff
@@ -56,6 +59,285 @@ def __iter__(self):
+
+    def __call__(
+        self, batch: "pyarrow.Table"
+    ) -> Union[
+        "torch.Tensor",
+        List["torch.Tensor"],
+        Dict[str, "torch.Tensor"],
+        Dict[str, List["torch.Tensor"]],
+    ]:
+        """Convert an Arrow batch to PyTorch tensors.
+
+        Args:
+            batch: PyArrow Table to convert
+
+        Returns:
+            Collated PyTorch tensors, can be:
+            - A single tensor
+            - A list of tensors
+            - A dict of column name to tensor
+            - A dict of column name to list of tensors
+        """
+        from ray.air._internal.torch_utils import (
+            arrow_table_to_tensors,
+        )
+
+        return arrow_table_to_tensors(batch, dtypes=self.dtypes, device=self.device)
+
+
+@DeveloperAPI
+class DefaultNumpyCollateFn(NumpyBatchCollateFn):
+    """Default collate function for converting Numpy batches to PyTorch tensors."""
+
+    def __init__(
+        self,
+        dtypes: Optional[Union["torch.dtype", Dict[str, "torch.dtype"]]] = None,
+        device: Optional[str] = None,
+    ):
+        """Initialize the collate function.
+
+        Args:
+            dtypes: Optional torch dtype(s) for the tensors
+            device: Optional device to place tensors on
+        """
+        super().__init__(dtypes=dtypes, device=device)
+
+    def __call__(self, batch: "pandas.DataFrame") -> "CollatedData":
+        """Convert a Pandas batch to PyTorch tensors.
+
+        Args:
+            batch: Pandas batch to convert
+
+        Returns:
+            Collated PyTorch tensors
+        """
+        from ray.air._internal.torch_utils import (
+            convert_ndarray_batch_to_torch_tensor_batch,
+        )
+
+        return convert_ndarray_batch_to_torch_tensor_batch(
+            batch, dtypes=self.dtypes, device=self.device
+        )
+
+
+@DeveloperAPI
+class DefaultPandasCollateFn(PandasBatchCollateFn):
+    """Default collate function for converting Pandas batches to PyTorch tensors."""
+
+    def __init__(
+        self,
+        dtypes: Optional[Union["torch.dtype", Dict[str, "torch.dtype"]]] = None,
+        device: Optional[str] = None,
+    ):
+        """Initialize the collate function.
+
+        Args:
+            dtypes: Optional torch dtype(s) for the tensors
+            device: Optional device to place tensors on
+        """
+        super().__init__(dtypes=dtypes, device=device)
+
+    def __call__(self, batch: "pandas.DataFrame") -> "CollatedData":
+        """Convert a Pandas batch to PyTorch tensors.
+
+        Args:
+            batch: Pandas batch to convert
+
+        Returns:
+            Collated PyTorch tensors
+        """
+        from ray.air._internal.torch_utils import (
+            convert_pandas_batch_to_torch_tensor_batch,
+        )
+
+        return convert_pandas_batch_to_torch_tensor_batch(
+            batch, dtypes=self.dtypes, device=self.device
+        )
+
+
+@DeveloperAPI
+class DefaultFinalizeFn:
```

Comment:
since we don't expose finalize_fn as an arg for now. this can be a regular function (not a callable class)